tools/include/uapi/linux/bpf.h

```diff
@@ -2890,6 +2890,111 @@ union bpf_attr {
+ *	Description
+ *		Pushes a light userdata onto the Lua stack.
+ *		Userdata represent C values in Lua.
+ *		A light userdata represents a pointer, a void*.
+ *		It is a value (like a number): you do not create it,
+ *		it has no individual metatable, and it is not collected
+ *		(as it was never created).
+ *		A light userdata is equal to "any" light userdata with
+ *		the same C address.
+ *
+ * void bpf_lua_pushlstring(void *ctx, const char *s, size_t len)
+ *	Description
+ *		Pushes the string pointed to by s with size len onto the stack.
+ *		Lua makes (or reuses) an internal copy of the given string,
+ *		so the memory at s can be freed or reused immediately after the
+ *		function returns.
+ *		The string can contain any binary data, including embedded zeros.
+ *
+ * void bpf_lua_pushmap(void *ctx, void *map)
+ *	Description
+ *		Pushes a BPF map onto the Lua stack
+ *
+ * void bpf_lua_pushskb(void *ctx)
+ *	Description
+ *		Pushes an SKB structure onto the Lua stack
+ *
+ * void bpf_lua_pushstring(void *ctx, const char *s)
+ *	Description
+ *		Pushes the zero-terminated string pointed to by s onto the stack.
+ *		Lua makes (or reuses) an internal copy of the given string,
+ *		so the memory at s can be freed or reused immediately after the
+ *		function returns.
+ *
+ * int bpf_lua_toboolean(void *ctx, int index)
+ *	Description
+ *		Converts the Lua value at the given index to a C
+ *		boolean value (0 or 1)
+ *	Return
+ *		1 if the value in the given index of the Lua stack is
+ *		different from from false or null, otherwise returns 0
+ *
+ * int bpf_lua_tointeger(void *ctx, int index)
+ *	Description
+ *		Converts the Lua value at the given index of the Lua stack
+ *		to the signed integral type lua_Integer.
+ *	Return
+ *		The converted Lua value at the given index, if the value is
+ *		convertible to an integer(see the Lua manual for more details
+ *		on type conversion); otherwise returns 0
+ *
+ * void bpf_lua_newpacket(void *ctx, int offset)
+ *	Description
+ *		Create new luaunpack user data buffer pointing to
+ *		the captured packet at the specified offset
+ *
+ * void bpf_lua_tostring(void *ctx, const char *str, u32 size, int index)
+ *	Description
+ *		Converts the Lua value at the given index of the Lua
+ *		stack to a C string and copies size bytes of it to
+ *		value pointed by str
+ *	Return
+ *		1 if the value at the given index of the Lua stack is a
+ *		string; otherwise it returns 0
+ *
+ * int bpf_lua_type(void *ctx, int index)
+ *	Description
+ *		Obtains the type of the Lua value at the given index
+ *		of the Lua stack
+ *
+ *	Return
+ *		Type of the value in the given valid index,
+ *		or LUA_TNONE for a non-valid (but acceptable) index.
+ *		The types returned by lua_type are coded by the
+ *		following constants defined in lua.h: LUA_TNIL (0),
+ *		LUA_TNUMBER, LUA_TBOOLEAN, LUA_TSTRING, LUA_TTABLE,
+ *		LUA_TFUNCTION, LUA_TUSERDATA, LUA_TTHREAD, and
+ *		LUA_TLIGHTUSERDATA.
```

Comment:
we don't need this long description.. you can just refer to the manual or lua.h itself as you're doing already..